presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -250,6 +250,7 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
             if (catalogProperties.isEmpty()) {
                 continue;
             }
+            accessControl.checkCanAccessCatalog(transactionId, getIdentity(), catalogName);
```

Comment:
The error here should be "Catalog does not exist", probably, since I'd imagine you can get here when specifying the catalog for the CLI, etc.